releasenotes/notes/add-cmap-componets-7ed56cdf294150f1.yaml

```diff
@@ -0,0 +1,13 @@
+---
+features:
+  - |
+    Added support to the :class:`~.CouplingMap` object to have a disjoint
+    connectivity. Previously, a :class:`~.CouplingMap` could only be
+    constructed if the graph was connected. This will enable using
+    :class:`~.CouplingMap` to represent hardware with disjoint qubits.
```

Comment:
It's not necessarily on separate chips though is it? Not that it's a good example but if you had a qubit that didn't have any connectivity to the others on the chip this would also be a separate component. The primary motivation for this is multichip devices, but it's not exclusively for that use case.

Reply:
This is a good point (and part of what I'm realizing I missed in writing https://github.com/Qiskit/qiskit-terra/pull/9710#discussion_r1127888460 ), there is a case where someone may want to intentionally subdivide a single connected device to emulate a larger LOCC device. That said, agree that this or other cases where this might come up (subdividing coupling graphs due to faulty qubits/gates, if we're interested in pursuing that) aren't the primary motivation here, so it would be good to highlight that in the release note.

Reply:
Well it's more I was thinking in this context someone could conceivably build a single chip device that doesn't have full connectivity (either because of manufacturing faults or by design). How about something like:

```suggestion
    :class:`~.CouplingMap` to represent hardware with disjoint qubits, such as hardware
    with qubits on multiple separate chips?
```